flutter-for-android.md

```diff
@@ -0,0 +1,269 @@
+
+[code]
+
+This is where you would typically do network or database calls.
+
+With regards to AsyncTask which follows a three step model onPreExecute, doInBackground and onPostExecute, doing a similar model in Flutter is much simpler. You can just write non-blocking code as it is, like you would in onPreExecute and onPostExecute and to run code asynchronously you would call an async function
+
+[code]
+
+onPostExecute typically takes the value from the result of the doInBackground task and updates the UI, in Flutter as you can see above you would just take the value returned from the await’ed function and update the UI by calling setState.
+
+However there are times where you may be processing a large amount of data and your UI will hang. 
+
+To get around this you can run code in the background. See <u>*How do I run parallel code on multi-core devices? I.e AsyncTask.execute(AsyncTask.THREAD_POOL_EXECUTOR) or run code in a background thread?*</u>
+
+
+
+**How do I run parallel code on multi-core devices? I.e AsyncTask.execute(AsyncTask.THREAD_POOL_EXECUTOR) or run code in a background thread?**
+
+Since Flutter uses a single threaded execution model, the code you write all executes on a single core on the mobile device. 
+
+However it is possible to take advantage of multiple CPU cores to do long running or computationally intensive tasks. This is done by using Isolates.
+
+Isolates are a separate execution thread that run and do not share any memory with the main execution thread. This means you can’t just access variables from the main thread or update your UI by calling setState. 
+
+Lets see an example of a simple Isolate and how you can communicate and share data back to the main thread to update your UI.
+
+[code]
+
+While it’s no golden rule, try to keep your Isolates to the number of cpu cores on the device as having too many will cause your execution to slow down.
+
+
+
+**How do I make network calls on Android and update the UI?**
+
+Making a network call in Flutter is very easy when you use the popular "http" package from https://pub.dartlang.org/packages/http
+
+You can use it by adding it to your dependecies in pubspec.yaml
+
+[code]
+
+Then to make a network call, for example to this JSON GIST on GitHub you can call
+
+[code]
+
+Once you have the result you can tell Flutter to update it's state by calling setState like so
+
+[code]
+
+Which will update your UI with the result from your network call.
+
+
+
+**How do I show progress dialogs on android when there is a task that is running?**
+
+Before you call your long running task, you can show a Dialog to let the user know there is processing happening. This can be done by rendering a Dialog widget. You can show the Dialog programatically by controlling it's rendering through a boolean and telling Flutter to update it's state just before your long running task.
```

Comment:
unless it really is dialog?

Maybe also use the word "indicator" here, even if the actual widget is Dialog.